test/index.js

```diff
@@ -15,32 +18,42 @@ const utils = {
   },
 }
 
-afterEach(() => {
+// Temporary switch cwd
+const changeCwd = function (cwd) {
+  const originalCwd = process.cwd()
+  process.chdir(cwd)
+  return process.chdir.bind(process, originalCwd)
+}
+
+// Switch cwd to a fixture directory
+const useFixture = function (fixtureName) {
+  const fixtureDir = `${FIXTURES_DIR}/${fixtureName}`
+  const restoreCwd = changeCwd(fixtureDir)
+  return { restoreCwd, fixtureDir }
+}
+
+// In each test, we change cwd to a temporary directory.
+// This allows us not to have to mock filesystem operations.
+beforeEach(async () => {
+  const { path, cleanup } = await getTmpDir({ unsafeCleanup: true })
+  const restoreCwd = changeCwd(path)
+  Object.assign(this, { cleanup, restoreCwd })
+})
+
+afterEach(async () => {
   utils.build.failBuild.mockReset()
   utils.run.command.mockReset()
   jest.clearAllMocks()
   jest.resetAllMocks()
+
+  this.restoreCwd()
+  await this.cleanup()
```

Comment:
what does cleanup do and why is it async?

Reply:
Added some comment :+1: